### ISSUE TITLE
The AUC metric is very low.

### ISSUE THREAD

**Issue by lxy21214** [BlueGhostYi/NIE-GCN#2]
We have evaluated your method on several benchmark datasets, including Amazon-Book, Yelp2018, MovieLens, and LastFM. In addition to commonly used evaluation metrics, we also measured its performance in terms of AUC.

<html xmlns:v="urn:schemas-microsoft-com:vml"
xmlns:o="urn:schemas-microsoft-com:office:office"
xmlns:x="urn:schemas-microsoft-com:office:excel"
xmlns="http://www.w3.org/TR/REC-html40">

<head>

<meta name=ProgId content=Excel.Sheet>
<meta name=Generator content="Microsoft Excel 15">
<link id=Main-File rel=Main-File
href="file:///C:/Users/ADMINI~1/AppData/Local/Temp/msohtmlclip1/01/clip.htm">
<link rel=File-List
href="file:///C:/Users/ADMINI~1/AppData/Local/Temp/msohtmlclip1/01/clip_filelist.xml">
<style>

</style>
</head>

<body link="#0563C1" vlink="#954F72">

AUC | Last.FM | Amazon Book | MovieLens-20M | Yelp2018
-- | -- | -- | -- | --
MF | 0.8678 | 0.909 | 0.9788 | 0.9535
LightGCN | 0.8925 | 0.9121 | 0.9869 | 0.9547
SGL | 0.8872 | 0.8804 | 0.9611 | 0.8765
CKE | 0.8672 | 0.9077 | 0.9788 | 0.9533
KGCN | 0.8352 | 0.7678 | 0.9744 | 0.7336
KGAT | 0.8832 | 0.9041 | 0.9866 | 0.9482
CKAN | 0.8646 | 0.8709 | 0.9776 | 0.893
KGIN | 0.8883 | 0.9142 | 0.9871 | 0.9573
KGCL | 0.8807 | 0.8859 | 0.9662 | 0.9006
KGRec | 0.8989 | 0.9172 | 0.9874 | 0.9525
NIE-GCN | 0.5582 | 0.5217 | 0.8531 | 0.5109

</body>

</html>

While your method achieves strong results in recall and NDCG, it performs significantly worse in AUC compared to other state-of-the-art approaches. Upon a thorough examination of the prediction outputs, we observe that your model tends to assign a predicted score of 0.5 to a large number of item-user pairs when estimating the relevance between a user and all items.

``` python
# This table presents the frequency distribution of relevance scores generated by the model \
# when predicting the relevance between a specific user and all items in the dataset.
# The dataset comprises a total of 3,846 items. After excluding the 11 items used for training \
# purposes, predictions were made for the remaining 3,835 items.
# Notably, approximately two-fifths (2/5) of the predicted relevance scores are equal to 0.5.
[(0.5, 1445), (0.508431077003479, 14), (0.5061877965927124, 13), (0.4960511028766632, 11), 
(0.49721813201904297, 10), (0.5010307431221008, 10), (0.5054226517677307, 9), 
(0.5108204483985901, 9), (0.49505332112312317, 8), (0.4997348487377167, 8), 
(0.5021339058876038, 8), (0.5071232914924622, 8), (0.5105937123298645, 8), 
(0.5113871693611145, 8), (0.5130565166473389, 8), (0.49854233860969543, 7), 
(0.5020648241043091, 7), (0.5047838091850281, 7), (0.5099355578422546, 7), 
(0.5111484527587891, 7), (0.5117629766464233, 7), (0.5129024386405945, 7), 
(0.513438880443573, 7), (0.5137540102005005, 7), (0.5139468312263489, 7), 
(0.5141950845718384, 7), (0.5205018520355225, 7), (0.4894217252731323, 6), 
(0.4902563691139221, 6), (0.5009394288063049, 6), (0.5089254379272461, 6), ...]
```

In addition, we plotted the ROC curve for each user, and overlaid the ROC curves of all users within the same epoch to generate a "ROC density map". The evolution of this map across epochs is shown in the following video:

https://github.com/user-attachments/assets/ed11d97d-0c14-40b5-aad2-a3833f59abd7

(from epoch 1 to epoch 3000, ploted every 10 epoches)

<img width="640" height="480" alt="The Grapha of mean ROC curve after 2100 epoches" src="https://github.com/user-attachments/assets/9d7cfbd8-422e-4a50-8242-70e601b42562" />

(Mean ROC Curve after 2100 epoches)

**Comment by lxy21214** [comment]
Okay, I get the reason.
The default value of GCNLayer is 1, which makes the model too simple to avoid overfitting.
